contracts/common/MultiSig.sol

```diff
@@ -715,19 +743,49 @@ contract MultiSig is Initializable, UUPSUpgradeable, UsingRegistryNoStorage {
         address[] calldata destinations,
         uint256[] calldata values,
         bytes[] calldata payloads
-    ) external {
-        address governanceAddress = address(getGovernance());
-
-        if (msg.sender != governanceAddress) {
-            revert SenderNotGovernance(msg.sender);
-        }
-
+    ) external onlyGovernance {
         for (uint256 i = 0; i < destinations.length; i++) {
             bytes memory returnData = ExternalCall.execute(destinations[i], values[i], payloads[i]);
             emit GovernanceTransactionExecuted(i, returnData);
         }
     }
 
+    /**
+     * @notice Sets the address of the Pauser contract that will be used to
+     * pause StakedCelo protocol contracts.
+     * @param __pauser The Pauser address to set.
+     */
+    function setPauser(address __pauser) external onlyWallet {
+        _pauser = IPauser(__pauser);
+        _setPauser(__pauser);
+    }
+
+    /**
+     * @notice Pauses the given StakedCelo protocol contracts.
+     * @param contracts The list of contracts to pause.
+     * @dev Can be called by any one of the MultiSig signers to immediately
+     * pause contracts. To be used to mitigate damage if a vulnerability is
+     * found/exploited.
+     */
+    function pauseContracts(address[] calldata contracts) external ownerExists(msg.sender) {
+        for (uint256 i = 0; i < contracts.length; i++) {
+            _pauser.pause(contracts[i]);
+        }
+    }
+
+    /**
+     * @notice Unpauses the given StakedCelo protocol contracts.
+     * @param contracts The list of contracts to unpause.
+     * @dev Can be called by Celo Governance (via hotfix or referendum). To be
+     * used after contracts have been paused with `pauseContracts`, and the
+     * related vulnerability have been patched.
+     */
+    function unpauseContracts(address[] calldata contracts) external onlyGovernance {
```

Comment:
So looking at this function, it seems that the intention is to only allow governance to unpause contracts once they are paused. 

However, if `pauseContracts()` is initially called without the multisig address being in the list, it is still possible for the multisig to unpause the other contracts following a proposal.

Is this intentional?

Reply:
Yes, I think this is fine. `pauseContracts` can always be called by any owner, so if it remaining unpaused is a problem, that can be easily corrected. While paused, `executeProposal` is locked, so even if a malicious proposal is submitted, its execution can be prevented by pausing.